src/commands/fipo.ts

```diff
@@ -56,137 +79,120 @@ registerCommand({
 
       // the array is still needed though, to not start a lot of timeouts
 
-      let alreadyDone = await (() => {
-        return new Promise((resolve) => {
-          db.get(
-            "SELECT * FROM fipos WHERE date = ?",
-            [
-              getAsStringDateWithCorrectTimezoneForReal(
-                new Date(fipo.createdTimestamp),
-              ),
-            ],
-            (err, row) => {
-              if (err) {
-                console.error(err);
-                return;
-              }
-
-              resolve(row != null);
-            },
-          );
-        });
-      })();
+      const alreadyDone: object | undefined = db.prepare(
+        "SELECT * FROM fipos WHERE date = ?",
+      ).get(
+        getAsStringDateWithCorrectTimezoneForReal(
+          new Date(fipo.createdTimestamp),
+        ),
+      );
 
       if (alreadyDone) {
         console.log({ fipoAlreadyDone: alreadyDone });
         return;
       }
 
+      if (
+        !(message.channel instanceof TextChannel)
+        || message.channel.id !== "789249810032361508"
+      ) {
+        message.reply(
+          "je kan alleen fipo doen in <#789249810032361508> makker",
+        );
+        return;
+      }
+
       message.channel.send("W00t " + fipo.author.toString() + "!");
 
-      db.run("INSERT INTO fipos (discord_id, date) VALUES (?, ?)", [
-        fipo.author.id,
-        getAsStringDateWithCorrectTimezoneForReal(
-          new Date(fipo.createdTimestamp),
-        ),
-      ]);
+      db.prepare("INSERT INTO fipos (discord_id, date) VALUES (?, ?)")
+        .get(
+          fipo.author.id,
+          getAsStringDateWithCorrectTimezoneForReal(
+            new Date(fipo.createdTimestamp),
+          ),
+        );
     }, 1000);
   },
-});
+};
 
-registerCommand({
-  name: "Fipo Statistics",
-  command: "fipostats",
+export const fipoStats: Command = {
+  name: "fipostats",
+  command: ".fipostats",
   description: "Check the fipo stats",
+  showInHelp: true,
+  match: (message: Message) => message.content === ".fipostats",
+  execute: async (message: Message): Promise<void> => {
+    if (!message.guild) return;
+
+    // for some reason unbeknownst to me, this is an array
+    const rawFipoEntries: Record<string, number | string>[] = db.prepare(
+      "SELECT * FROM fipos",
+    ).all();
+
+    const fipoEntries: { discord_id: string; date: string }[] = [];
+
+    for (const rawFipoEntry of rawFipoEntries) {
+      fipoEntries.push({
+        discord_id: String(rawFipoEntry.discord_id),
+        date: String(rawFipoEntry.date),
+      });
+    }
 
```

Comment:
This way, we avoid a lot of unneccesary calculations:

```suggestion
    const fipo = db.sql`
      SELECT DISTINCT(discord_id), COUNT(*) as fipos
      FROM fipos
      GROUP BY discord_id
      ORDER BY fipos DESC
      LIMIT 50
    ` as { discord_id: string, fipos: number }[];

    const members = await message.guild.members.fetch({ user: fipo.map(e => e.discord_id) });
    const fipoStats: [string, number][] = [];
    for (const { discord_id, fipos } of fipo) {
      const displayName = members.get(discord_id)?.displayName
      if (!displayName) continue;
      fipoStats.push([ displayName, fipos ]);
    }

    const longestUsername = fipoStats.reduce((acc, [displayName]) => Math.max(acc, displayName.length), 0);

    let returnMessage: string = "# fipostats\n```\n";
    for (const [displayName, fipos] of fipoStats) {
      returnMessage += `${displayName}${" ".repeat(longestUsername + 4 - displayName.length)}: ${fipos}\n`;
    }
    message.reply(returnMessage + "```");
```